### ISSUE TITLE
Coverage is broken in Intern 4.5+

### ISSUE THREAD

**Issue by jason0x43** [theintern/intern#1158]
In at least some scenarios, coverage in Intern 4.5+ is broken.

Example:

```ts
// src/app.ts
export function sayHi(name: string) {
  return `Hi, ${name}!`;
}
```

```ts
// tests/unit/app.ts
import { sayHi } from '../../src/app';

const { suite, test } = intern.getPlugin('interface.tdd');
const { assert } = intern.getPlugin('chai');

suite('app', () => {
  test('sayHi', () => {
    assert.equal(sayHi('Bob'), 'Hi, Bob!');
  });
});
```

```js
// intern.json
{
  "coverage": ["src/*.js"],
  "suites": "tests/unit/app.js"
}
```

With Intern 4.8.5 on Node 10.15.3:

```
$ npx intern debug
DEBUG: Using default loader
DEBUG: Instrumenting /Users/jason/Temp/intern/intern4-test/src/app.js
DEBUG: Instrumenting /Users/jason/Temp/intern/intern4-test/src/app.js
(ノಠ益ಠ)ノ彡┻━┻
TypeError: Cannot read property 'f' of undefined
  at Object.<anonymous> @ src/app.js:1:0
  at Module._compile @ internal/modules/cjs/loader.js:701:30
```

This same example with Intern 4.3.6:

```
$ npx intern debug
DEBUG: Using default loader
DEBUG: Instrumenting /Users/jason/Temp/intern/intern4-test/src/app.js
DEBUG: Loaded suites: ["tests/unit/app.js"]
✓ node - app - sayHi (0.001s)

Total coverage
----------|----------|----------|----------|----------|-------------------|
File      |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
----------|----------|----------|----------|----------|-------------------|
All files |       70 |       40 |      100 |    77.78 |                   |
 app.js   |       70 |       40 |      100 |    77.78 |               6,7 |
----------|----------|----------|----------|----------|-------------------|
TOTAL: tested 1 platforms, 1 passed, 0 failed
```

Note that the app file is instrumented twice in 4.8.5. That's not good.

**Comment by jason0x43** [comment]
4.8.0 exhibits the broken behavior.

**Comment by jason0x43** [comment]
4.4.3 was the last version that doesn't exhibit the double instrumenting behavior.

**Comment by jason0x43** [comment]
The istanbul dependencies were updated in Intern 4.5.0. In particular, `istanbul-lib-coverage` was updated from 1.1.0 to 2.0.1. Possibly that's the source of the issue.

**Comment by jason0x43** [comment]
The problem is with `istanbul-lib-hook`. Downgrading to 1.2.2 works. Version > 1.2.2 will run both the "runInThisContext" and "require" hooks when importing files.

**Comment by jason0x43** [comment]
The root problem is that, prior to the use of istanbul-lib-hook@2 in Intern 4.5, the `runInThisContext` hook was never executed (at least with Node 10+) because istanbul wasn't using the correct API for `runInThisContext` (https://github.com/istanbuljs/issues#89). The issue doesn't occur for Intern's self tests.

**Comment by jason0x43** [comment]
Resolved in 1160f8e. Intern 5-pre did not have this issue.
